### DISCUSSION TITLE
OSIC Microelectronics / GDSFactory Best integration approach

### DISCUSSION THREAD

**Discussion by daquintero** [gdsfactory/gdsfactory#1780] [Ideas]
Following on from https://github.com/gdsfactory/gdsfactory/issues/1720

I have already been working on a gdsfactory plugin to integrate it, that is still green which is why I hadn't mentioned yet, https://github.com/daquintero/piel . The idea is to provide a set of interconnection functions between gdsfactory core and the existing [IIC-OSIC-TOOLS](https://github.com/iic-jku/iic-osic-tools) workflow. I thought about integrating it directly into gdsfactory, ie through this eg local branch https://github.com/daquintero/gdsfactory/tree/openlane_integration .

The problem is an environment one. The configuration for OS microelectronic design tools is a lot more involved than for just GDSFactory, and there is a lot more interaction between the design files and environment flags. It might add a bit too much overhead to add it as a dependency directly into gdsfactory than as an external plugin package like `piel`. There are a lot more dependencies and a docker environment configuration that might be too much overhead for a default gdsfactory installation for most users. What do you think? I'll make another issue to discuss as this is a bit deviating from this main Cadence plugin issue.

**Comment by daquintero** [comment]
Another thing I would like to add is an idea of `AbstractComponents` that can be part of a `gdsfactory` physical netlist but not necessarily an actual design, because this would enable us to simulate multi-stack systems ie. coaxial interconnection and so on.

**Comment by joamatab** [comment]
This looks great Dario

@hpretl

**Comment by daquintero** [comment]
I would love to integrate Dan Fritchman's many projects into this plugin so that gdsfactory can talk with his design tools, maybe if I get enough time I'll do it too

**Comment by joamatab** [comment]
https://github.com/gdsfactory/gplugins/pull/80
